### PR TITLE
AC-666: Create phpcs static check for CopyrightTest for php files

### PR DESCRIPTION
**Pull request by eliseacornejo**

### CODE REVIEW

**Review by svera — CHANGES_REQUESTED** [review]
Shouldn't we use the Adobe copyright instead?

File: Magento2/Tests/Legacy/CopyrightUnitTest.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+     * @inheritdoc
+     */
+    public function getWarningList($testFile = ''): array
+    {
+        if ($testFile === 'CopyrightUnitTest.4.inc') {
+            return [];
+        }
+        
+        return [
+            null => 1,
```
Comment (loginesta) [inline Magento2/Tests/Legacy/CopyrightUnitTest.php:30]: just want to confirm why... are you using `null` to indicate that the whole file is incorrect?

**Review by loginesta — APPROVED** [review]

File: Magento2/Tests/Legacy/CopyrightUnitTest.php (hunk 1 above)
Comment (eliseacornejo) [reply Magento2/Tests/Legacy/CopyrightUnitTest.php:30]: When I did, it I was thinking in the case you don't have the comment at all, but when you have it and it is not correct may be is better to add the line. I will do the changes.

**Comment by eliseacornejo** [comment]
I don't know if we should use the Adobe one, @sivaschenko do you know if we can change the copyright to the Adobe one?

File: Magento2/Sniffs/Legacy/CopyrightSniff.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $positionOpenTag = $phpcsFile->findPrevious(T_OPEN_TAG, $stackPtr - 1);
+       
+        if ($positionOpenTag === false) {
+            $positionComment = $phpcsFile->findNext(T_DOC_COMMENT_STRING, $stackPtr);
+            $contentFile = $phpcsFile->getTokens()[$positionComment]['content'];
+            $adobeCopyrightFound = preg_match(self::COPYRIGHT_ADOBE, $contentFile);
+            
+            if (strpos($contentFile, self::COPYRIGHT_MAGENTO_TEXT) === false || $adobeCopyrightFound === false) {
```
Comment (jcuerdo) [inline Magento2/Sniffs/Legacy/CopyrightSniff.php:40]: This is not working fine. It should be:
`if (strpos($contentFile, self::COPYRIGHT_MAGENTO_TEXT) !== false || $adobeCopyrightFound ) {return;}else{addWarning()}`

**Review by jcuerdo — CHANGES_REQUESTED** [review]

File: Magento2/Sniffs/Legacy/CopyrightSniff.php (hunk 2 above)
Comment (eliseacornejo) [reply Magento2/Sniffs/Legacy/CopyrightSniff.php]: Done

File: Magento2/Sniffs/Legacy/CopyrightSniff.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            if (strpos($contentFile, self::COPYRIGHT_MAGENTO_TEXT) !== false || $adobeCopyrightFound) {
+                return;
+            } else {
+                $phpcsFile->addWarningOnLine(
+                    'Copyright is missing or has wrong format',
+                    $phpcsFile->getTokens()[$positionComment]['line'],
+                    self::WARNING_CODE
+                );
+            }
+        }
```
Comment (sivaschenko) [inline Magento2/Sniffs/Legacy/CopyrightSniff.php:49]: I would be nice to flatten the code structure for better readability. Also handling the situation when $positionComment is false (no comments found) instantly (not trying to parse it)
```suggestion
        if ($phpcsFile->findPrevious(T_OPEN_TAG, $stackPtr - 1) !== false) {
            return;
        }

        $positionComment = $phpcsFile->findNext(T_DOC_COMMENT_STRING, $stackPtr);

        if ($positionComment === false) {
            $phpcsFile->addWarning(
                'Copyright is missing',
                $stackPtr,
                self::WARNING_CODE
            );
            return;
        }
        $content = $phpcsFile->getTokens()[$positionComment]['content'];
        $adobeCopyrightFound = preg_match(self::COPYRIGHT_ADOBE, $content);

        if (strpos($content, self::COPYRIGHT_MAGENTO_TEXT) !== false || $adobeCopyrightFound) {
            return;
        }

        $phpcsFile->addWarningOnLine(
            'Copyright is missing or has wrong format',
            $phpcsFile->getTokens()[$positionComment]['line'],
            self::WARNING_CODE
        );
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Great job @eliseacornejo ! Please see my comment.

File: Magento2/Sniffs/Legacy/CopyrightSniff.php (hunk 3 above)
Comment (eliseacornejo) [reply Magento2/Sniffs/Legacy/CopyrightSniff.php]: Thanks for your collaboration. Done!

**Review by sivaschenko — APPROVED** [review]

**Comment by eliseacornejo** [comment]
@magento import pr to magento-commerce/magento-coding-standard

**Comment by eliseacornejo** [comment]
@magento import pr to magento-commerce/magento-coding-standard

**Comment by magento-engcom-team** [comment]
@eliseacornejo the pull request successfully imported.

**Comment by obukhow** [comment]
Could you please modify this sniff to work only for Magento's namespace? We don't need any Adobe or Magento copyright in our code.

**Comment by fredden** [comment]
@obukhow this is being discussed in https://github.com/magento/magento-coding-standard/issues/275.

**Pull request merged**